### ISSUE TITLE
NewProcess does not return an error if process does not exist on Windows

### ISSUE THREAD

**Issue by dhaavi** [shirou/gopsutil#729]
**Describe the bug**

The docs for `NewProcess` state that `An error will be returned if the process does not exist.`
Running on Windows 10, `NewProcess` does _not_ return an error if called with an invalid PID.
(Works fine on Linux 4.9)

**To Reproduce**

```go
package main

import (
	"fmt"

	"github.com/shirou/gopsutil/process"
)

func main() {
	_, err := process.NewProcess(9999)
	if err == nil {
		fmt.Println("process exists!")
	}
}
```

**Expected behavior**
`NewProcess` should return an error.

**Environment (please complete the following information):**
 - [x] Windows: `Microsoft Windows [Version 10.0.17763.615]`
 - [x] Linux: `Linux 74b9ced6499d 4.9.125-linuxkit #1 SMP Fri Sep 7 08:20:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
 - [ ] Mac OS: [paste the result of `sw_vers` and `uname -a`
 - [ ] FreeBSD: [paste the result of `freebsd-version -k -r -u` and  `uname -a`]
 - [ ] OpenBSD: [paste the result of `uname -a`]

**Additional context**
Library version:
```
  name = "github.com/shirou/gopsutil"
  revision = "4c8b404ee5c53b04b04f34b1744a26bf5d2910de"
  version = "v2.19.6"
```

Cross-compiling for Windows on Linux using `GOOS=windows go build` in a dir with only the file pasted above.

```
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/user/.gocache"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/user/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/lib/go-1.12"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go-1.12/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build008209370=/tmp/go-build -gno-record-gcc-switches"
```

**Workaround**

```go
package main

import (
	"fmt"

	"github.com/shirou/gopsutil/process"
)

func main() {
	p, err := process.NewProcess(9999)
	if err == nil {
		_, err = p.Name() // edited
		if err == nil {
			fmt.Println("process exists!")
		}
	}
}
```

**Comment by Lomanic** [comment]
This check is only present on Linux currently (checking if `/proc/$PID` exists). Work in #713 will make it viable (without destroying performances as it would have before on some platforms, for example on Darwin) to check PID existence each time you call `process.NewProcess`.

**Comment by Lomanic** [comment]
Trying to call `process.Status()` on Windows is a bad idea to check if process exists on Windows [as it is not implemented](https://github.com/shirou/gopsutil/blob/507e2f7a534eb1f2b96cd8448f99261ccb3ac2bb/process/process_windows.go#L331) and returns a `common.ErrNotImplementedError`.

**Comment by dhaavi** [comment]
I see. Could you then update the documentation to reflect this behavior? The windows version of `NewProcess` is undocumented. (I must admit, I did not check it before raising the issue)

Ha! Good Catch! Thanks for that.
What would you recommend as a workaround until this can be fixed with #713?

**Comment by Lomanic** [comment]
> The windows version of NewProcess is undocumented.

Documentation is only present on linux because it's godoc.org default GOOS (last time I checked there was no way to change GOOS  via URL parameter on godoc.org) and because we have the same documentation for all the platforms.

You can check for process existence with [`process.PidExists()`](https://godoc.org/github.com/shirou/gopsutil/process#PidExists). This will be slow with current implementation of `process.Pids()` on darwin though (and on others that may call `ps` to list processes).

> Could you then update the documentation to reflect this behavior?

This will be fixed with #713 being merged and the test for process existence being added to `process.NewProcess()` ; it's been [like this for quite a long time](https://github.com/shirou/gopsutil/commit/a3bbd9e3cdb1e1c7e757b395e6e806fc4353e442), there is little value for me to modify the documentation when it will be fixed in the coming weeks.

**Comment by dhaavi** [comment]
> there was no way to change GOOS via URL parameter on godoc.org

I know - that is why i checked the source code. I am really annoyed by this behavior of godoc.org.
 
> there is little value for me to modify the documentation when it will be fixed in the coming weeks.

Perfect. Wasn't aware of that timeline.
